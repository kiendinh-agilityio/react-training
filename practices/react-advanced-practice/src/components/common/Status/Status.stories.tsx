import Status from '.';

export default {
  component: Status,
  title: 'Status',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const Active = {
  args: {
    value: 'Active',
  },
};

export const Inactive = {
  args: {
    value: 'Inactive',
  },
};
