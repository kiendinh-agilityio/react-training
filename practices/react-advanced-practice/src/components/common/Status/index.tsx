import { memo } from 'react';

// Import radix ui
import { Flex } from '@radix-ui/themes';

// Import components
import { Text } from '@/components/common';

interface StatusProps {
  value: string;
}

const Status = memo(({ value }: StatusProps) => (
  <Flex
    justify="center"
    align="center"
    className={`w-[70px] h-[25px] border rounded-lg ${
      value === 'Active' ? 'border-active bg-active' : 'border-inactive bg-inactive'
    }`}
  >
    <Text weight="bold" className="font-bold text-white" content={value} />
  </Flex>
));

export default Status;
