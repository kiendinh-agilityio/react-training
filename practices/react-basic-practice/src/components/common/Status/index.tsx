import { Paragraph } from '@/components/common';

interface StatusProps {
  value: string;
}

const Status = ({ value }: StatusProps) => (
  <div
    className={`flex justify-center items-center w-[70px] h-[25px] border rounded-lg font-helveticaBold font-bold text-white ${
      value === 'Active' ? 'border-active bg-active' : 'border-inactive bg-inactive'
    }`}
  >
    <Paragraph value={value} />
  </div>
);

export default Status;
