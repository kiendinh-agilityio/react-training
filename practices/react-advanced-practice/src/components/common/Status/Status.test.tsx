import { render } from '@testing-library/react';
import Status from '.';

describe('Status Component', () => {
  const renderComponent = (value: string) => render(<Status value={value} />);

  it('renders correctly with "Active" value', () => {
    const { container, getByText } = renderComponent('Active');
    expect(getByText('Active')).toBeInTheDocument();
    expect(container.firstChild).toHaveClass('border-active bg-active');
    expect(container).toMatchSnapshot();
  });

  it('renders correctly with "Inactive" value', () => {
    const { container, getByText } = renderComponent('Inactive');
    expect(getByText('Inactive')).toBeInTheDocument();
    expect(container.firstChild).toHaveClass('border-inactive bg-inactive');
    expect(container).toMatchSnapshot();
  });
});
