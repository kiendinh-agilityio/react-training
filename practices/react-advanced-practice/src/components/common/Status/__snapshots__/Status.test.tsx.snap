// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Component renders correctly with "Active" value 1`] = `
<div>
  <div
    class="rt-Flex rt-r-ai-center rt-r-jc-center w-[70px] h-[25px] border rounded-lg border-active bg-active"
  >
    <p
      class="rt-Text rt-r-weight-bold text-sm font-regular font-bold text-white"
    >
      Active
    </p>
  </div>
</div>
`;

exports[`Status Component renders correctly with "Inactive" value 1`] = `
<div>
  <div
    class="rt-Flex rt-r-ai-center rt-r-jc-center w-[70px] h-[25px] border rounded-lg border-inactive bg-inactive"
  >
    <p
      class="rt-Text rt-r-weight-bold text-sm font-regular font-bold text-white"
    >
      Inactive
    </p>
  </div>
</div>
`;
